python/lsst/pipe/base/graph.py

```diff
@@ -96,18 +135,67 @@ def __init__(self, iterable=None):
     def quanta(self):
         """Iterator over quanta in a graph.
 
+        Quanta are returned in unspecified order.
+
         Yields
         ------
         taskDef : `TaskDef`
             Task definition for a Quantum.
-        quantum : `Quantum`
+        quantum : `~lsst.daf.butler.Quantum`
             Single quantum.
         """
         for taskNodes in self:
             taskDef = taskNodes.taskDef
             for quantum in taskNodes.quanta:
                 yield taskDef, quantum
 
+    def orderedTaskNodes(self):
+        """Return topologically ordered task nodes.
+
+        Yields
+        ------
+        nodes : `QuantumGraphTaskNodes`
+        """
+        # Tasks in a graph are probably topologically sorted already but there
+        # is no guarantee for that. Just re-construct Pipeline and order tasks
+        # in a pipeline using existing method.
+        nodesMap = {id(item.taskDef): item for item in self}
+        pipeline = orderPipeline(Pipeline(item.taskDef for item in self))
+        for taskDef in pipeline:
+            yield nodesMap[id(taskDef)]
```

Comment:
If we don't have a concrete use case for this method, I would remove it; as implied by my earlier comments on the internal data structure for `QuantumGraph`, I think it implies and exposes an implementation that may not be something we want to keep.

Reply:
I't only used by `orderedQuanta()` now, I'll move it into that method.